pyuvsim/simsetup.py

```diff
@@ -1083,80 +1090,109 @@ def initialize_uvdata_from_params(obs_params):
+        integration_time=None, start_time=None, time_array=None, bls=None,
+        antenna_nums=None, antenna_names=None, polarization_array=None, no_autos=False,
+        redundant_threshold=None, write_files=True, path_out=None, complete=False,
+        **kwargs):
     """
     Setup a UVData object from keywords.
 
-    Args:
-        yaml_filename : str (optional)
-            Specify filename for yaml file to write out.
-            Defaults to obsparam.yaml
-        antenna_layout_filename : str (optional)
-            Path to csv file of antenna positions (see documentation for details).
-            Will not overwrite existing files.
-            Defaults to antenna_layout.csv
-        array_layout : dictionary (required if antenna_layout_filename not given)
-            dictionary, keys are integer antenna numbers, values are len-3 antenna positions
-            in ENU coordinates [meters].
-        antenna_names : list of str (optional)
-            If unset, antenna names are assigned as "%s" % antnum.
-        telescope_location : len-3 tuple
-            Telescope location on Earth in LatLonAlt coordinates [deg, deg, meters]
-        telescope_name : str
-            Name of telescope
-        Nfreqs : int
-            Number of frequency channels
-        start_freq : float
-            Starting frequency [Hz]
-        bandwidth : float
-            Total frequency bandwidth of spectral window [Hz]
-        channel_width: float
-            Frequency channel spacing [Hz]
-        freq_array : ndarray
-            frequency array [Hz], if this is specified, it supersedes Nfreqs, start_freq, bandwidth
-        Ntimes : int
-            Number of integration bins
-        integration_time : float
-            Width of time bins [seconds]
-        start_time : float
-            Time of the first integration bin [Julian Date]
-        time_array : ndarray
-            time array [Julian Date]. If this is specified it supersedes values of Ntimes, start_time and integration_time
-        bls : list
-            List of antenna-pair tuples for baseline selection
-        redundant_threshold: float
-            Redundant baseline selection tolerance for selection [meters]
-        antenna_nums : list
-            List of antenna numbers to keep in array
-        polarizations : list
-            List of polarization strings to insert into object
-        no_autos : bool
-            If True, eliminate all auto correlations
-        write_files : bool
-            If True, write out the parameter information to yaml files.
-        path_out : str (optional)
-            Path in which to place generated configuration files, if write_files is True.
-            Defaults to current directory.
-        kwargs : dictionary
-            Any additional valid UVData attribute to assign to object.
-
-    Returns:
-        UVData object with zeroed data_array
+    Parameters
+    ----------
+    yaml_filename : str (optional)
```

Comment:
For added clarity, perhaps you change this to something line output_yaml_file?  Just to make it painfully obvious that this method doesn't accept input yaml files to initialize a uvdata object, that's another method.

Reply:
Good idea. I've changed it.